common/ingressgateway/ingressgateway.go

```diff
@@ -0,0 +1,98 @@
+package ingressgateway
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net"
+	"net/http"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+
+	"golang.org/x/net/context"
+)
+
+const (
+	// This environmental variable must hold FQDN of ingress gateway which client should access
+	ServiceNameEnv = "INGRESSGATEWAY_FQDN"
+)
+
+// Dialer is an interface which abstracts net.Dialer.
+type Dialer interface {
+	DialContext(ctx context.Context, network, address string) (net.Conn, error)
+}
+
+// clientCreator abstracts interactions with outside world for easy testing.
+type clientCreator struct {
+	Getenv        func(string) string
+	LookupHost    func(string) ([]string, error)
+	GetMinikubeIP func() (string, error)
+	Dialer        Dialer
+}
+
+// Default returns clientCreator ready for production use.
+func Default() *clientCreator {
+	return &clientCreator{
+		Getenv:        os.Getenv,
```

Comment:
Consider being more specific: This is used only to fetch ingress FQDN, so perhaps: `IngressFQDN func() string` is better than very generic `os.Getenv`

Reply:
I would agree with `IngressFQDN func() string` but the signature is different. It is `func(string) string`. I could change that to:
```go
IngressFQDN: func() { return os.Getenv(ServiceNameEnv) }
```
but in this approach we wouldn't be able to test if code looks up valid environmental variable. Comparing those two I prefer to have it dirty, but better tested.

Reply:
I will apply the suggestion. Also there will be new test to cover getting environment variable.